File: nativelib/src/main/scala/scala/scalanative/posix/sys/ioctl.scala
package scala.scalanative.posix.sys

import scalanative.native._

@extern
object ioctl {

  @name("scalanative_ioctl")
  def ioctl(fd: CInt, request: CLongInt, argp: Ptr[Byte]): CInt = extern

  @name("scalanative_FIONREAD")
  def FIONREAD: CLongInt = extern

  @name("scalanative_SIOCGIFCONF")
  def SIOCGIFCONF: CLongInt = extern

  @name("scalanative_SIOCGIFFLAGS")
  def SIOCGIFFLAGS: CLongInt = extern

  @name("scalanative_SIOCGIFHWADDR")
  def SIOCGIFHWADDR: CLongInt = extern

  @name("scalanative_SIOCGIFMTU")
  def SIOCGIFMTU: CLongInt = extern
}

File: nativelib/src/main/resources/posix/sys/ioctl.c
#include <sys/ioctl.h>

int scalanative_ioctl(int fd, long int request, void *argp) {
    return ioctl(fd, request, argp);
}

long int scalanative_FIONREAD() { return FIONREAD; }

long int scalanative_SIOCGIFCONF() { return SIOCGIFCONF; }
long int scalanative_SIOCGIFFLAGS() { return SIOCGIFFLAGS; }
long int scalanative_SIOCGIFHWADDR() { return SIOCGIFHWADDR; }
long int scalanative_SIOCGIFMTU() { return SIOCGIFMTU; }
